concent_api/core/message_handlers.py

```diff
@@ -1354,28 +1356,28 @@ def handle_send_subtask_results_verify(
             reason=message.concents.ServiceRefused.REASON.InvalidRequest,
         )
 
-    if Subtask.objects.filter(  # pylint: disable=no-member
-        subtask_id=compute_task_def['subtask_id'],
-        state__in=[
-            Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER.name,  # pylint: disable=no-member
-            Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name,     # pylint: disable=no-member
-        ]
-    ).exists():
-        return message.concents.ServiceRefused(
-            reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
-        )
+    try:
+        subtask = Subtask.objects.select_for_update().get(subtask_id=compute_task_def['subtask_id'])
+
+        if subtask.state_enum in[
+            Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER,
+            Subtask.SubtaskState.ADDITIONAL_VERIFICATION,
+        ]:
```

Comment:
This `if` and the one below should be in the `else` clause.

Reply:
done